components/family-studies/migrations/src/main/java/org/phenotips/studies/family/migrations/PhenotipsFamilyMigrations.java

```diff
@@ -104,38 +104,39 @@
+        XWikiContext xcontext, Session hsession)
     {
-        this.context = xcontext;
-        this.session = hsession;
-        long nextId = getLastUsedId() + 1;
-        String nextStringId = String.format("%s%07d", FAMILY_PREFIX, nextId);
-
-        EntityReference newFamilyRef = new EntityReference(nextStringId, EntityType.DOCUMENT, Family.DATA_SPACE);
-        XWikiDocument newFamilyXDocument = this.context.getWiki().getDocument(newFamilyRef, this.context);
-        if (!newFamilyXDocument.isNew()) {
-            throw new IllegalArgumentException("The new family id was already taken.");
-        }
-
-        this.setOwner(newFamilyXDocument, patientXDoc);
-        this.setFamilyObject(newFamilyXDocument, patientXDoc, nextId);
-        this.setPedigreeObject(newFamilyXDocument, pedigreeData, pedigreeImage);
-        this.setPermissionsObject(newFamilyXDocument, patientXDoc);
-        this.setVisibilityObject(newFamilyXDocument, patientXDoc);
-        this.setCollaborators(newFamilyXDocument, patientXDoc);
-
-        newFamilyXDocument.setAuthorReference(patientXDoc.getAuthorReference());
-        newFamilyXDocument.setCreatorReference(patientXDoc.getCreatorReference());
-        newFamilyXDocument.setContentAuthorReference(patientXDoc.getContentAuthorReference());
-        newFamilyXDocument.setParentReference(this.familyParentReference);
+        try {
+            this.context = xcontext;
+            this.session = hsession;
+            long nextId = getLastUsedId() + 1;
+            String nextStringId = String.format("%s%07d", FAMILY_PREFIX, nextId);
+
+            EntityReference newFamilyRef = new EntityReference(nextStringId, EntityType.DOCUMENT, Family.DATA_SPACE);
+            XWikiDocument newFamilyXDocument = this.context.getWiki().getDocument(newFamilyRef, this.context);
+            if (!newFamilyXDocument.isNew()) {
+                throw new IllegalArgumentException("The new family id was already taken.");
+            }
 
-        return newFamilyXDocument;
+            this.setOwner(newFamilyXDocument, patientXDoc);
+            this.setFamilyObject(newFamilyXDocument, patientXDoc, nextId);
+            this.setPedigreeObject(newFamilyXDocument, pedigreeData, pedigreeImage);
+            this.setPermissionsObject(newFamilyXDocument, patientXDoc);
+            this.setVisibilityObject(newFamilyXDocument, patientXDoc);
+            this.setCollaborators(newFamilyXDocument, patientXDoc);
+
+            newFamilyXDocument.setAuthorReference(patientXDoc.getAuthorReference());
+            newFamilyXDocument.setCreatorReference(patientXDoc.getCreatorReference());
+            newFamilyXDocument.setContentAuthorReference(patientXDoc.getContentAuthorReference());
+            newFamilyXDocument.setParentReference(this.familyParentReference);
+
+            return newFamilyXDocument;
+        } catch (Exception ex) {
+            return null;
```

Comment:
adding log to show `IllegalArgumentException("The new family id was already taken.")` would be bonus, otherwise won't be seen